### PR TITLE
Add explicit types to parameters

### PR DESCRIPTION
**Pull request by mmachatschek**
These routes receive arrays (as of the documention) as parameters and therefore should enforce the array type to these specific routes.

### CODE REVIEW

**Review by curquiza — APPROVED** [review]
LGTM, thanks a lot @mmachatschek! :)

We wait for the review of @bidoubiwa before any merge 🙂

**Review by bidoubiwa — APPROVED** [review]
Hello, thanks a lot for this PR! 🔥

I have a question. I noticed in other files that typing was not always done. For example
```php
   public function getDocument($documentId)
    {
        $this->assertValidDocumentId($documentId);

        return $this->http->get(self::PATH.'/'.$this->uid.'/documents/'.$documentId);
    }
```

I'm not at all expecting you to fix this! I was just wondering if maybe there is a reason why these have no types?

**Comment by mmachatschek** [comment]
Hi @bidoubiwa

The thing is that if we would type these as string, we would break implementations that pass in documents with integer ids.

I looked at most places and found that either docblocks (which are more applicable here) are missing, incomplete or missing. I think I can send in another PR trying to cover all docblocks and types. Maybe I find more, where it makes sense to add specific types.

Cheers

**Comment by bidoubiwa** [comment]
Thanks for the explaination :) Very cool if you do that but in either case thanks!

**Comment by curquiza** [comment]
thanks @bidoubiwa and @mmachatschek!
bors merge

**Pull request merged**
